modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -69,35 +71,45 @@ object Enrich {
    * [[Assets.State.make]] downloads assets for the first time unconditionally during
    * [[Environment]] initialisation, then if `assetsUpdatePeriod` has been specified -
    * they'll be refreshed periodically by [[Assets.updateStream]]
+   *
+   * @param ordered indicates whether the events should be processed ordered or not
    */
-  def run[F[_]: Concurrent: ContextShift: Clock: Parallel](env: Environment[F]): Stream[F, Unit] = {
+  def run[F[_]: Concurrent: ContextShift: Clock: Parallel, A](env: Environment[F, A], ordered: Boolean): Stream[F, Unit] = {
     val registry: F[EnrichmentRegistry[F]] = env.enrichments.get.map(_.registry)
     val enrich: Enrich[F] = {
       implicit val rl: RegistryLookup[F] = env.registryLookup
-      enrichWith[F](registry, env.igluClient, env.sentry, env.metrics.enrichLatency)
+      enrichWith[F](registry, env.igluClient, env.sentry, env.metrics.enrichLatency, env.processor)
     }
 
+    val enrichPipe: Pipe[F, (A, Array[Byte]), (A, Result)] =
+      if(ordered)
+        _.parEvalMapUnordered(ConcurrencyLevel){ case (orig, bytes) => enrich(bytes).map((orig, _)) }
+      else
+        _.parEvalMap(ConcurrencyLevel){ case (orig, bytes) => enrich(bytes).map((orig, _)) }
```

Comment:
```suggestion
      in =>
        val parallel = if (ordered) in.parEvalMap(ConcurrencyLevel) else in.parEvalMapUnordered(ConcurrencyLevel) 
        parallel { case (orig, bytes) => enrich(bytes).map((orig, _)) }
```
 
Haven't checked if it compiles - you might need to help compiler to infer the type, but it highlights that `parEvalMapUnordered`/`parEvalMap` is the only difference.